### PR TITLE
[Subscription block] Add social followers toggle

### PR DESCRIPTION
**Pull request by TimBroddin**
#### Changes proposed in this Pull Request:
This PR adds a toggle to the Subscription block settings, that allows to toggle between including social followers (default) in the count, or excluding them.

CleanShot 2022-11-16 at 16 02 36@2x

Jetpack product discussion
pdDOJh-ZI-p2
pdDOJh-Yp-p2

Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
1. Apply this PR
2. The checkbox should be visible & toggling it should update the number in both editor & frontend.

To test with simple sites:
- On your sandbox:
  - `bin/jetpack-downloader test jetpack add/subscription-block-social-toggle`(don't checkout trunk)

To test with Atomic:
- Follow these instructions: pb5gDS-1rQ-p2
 
Also make sure that users see the same count when upgrading (standard behavior should be to include social followers)

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/subscribers.php [hunk 1]
```diff
@@ -72,6 +84,22 @@ public function get_subscriber_count( $request ) { // phpcs:ignore VariableAnaly
...
+	 *
+	 * @return array data object containing subscriber counts ['email_subscribers' => 0, 'social_followers' => 0]
+	 */
+	public function get_subscriber_counts() {
+		if ( ! Constants::is_defined( 'TESTING_IN_JETPACK' ) ) {
+			delete_transient( 'wpcom_subscribers_totals' );
+		}
+
+		$subscriber_info  = Jetpack_Subscriptions_Widget::fetch_subscriber_counts();
+		$subscriber_count = $subscriber_info['value'];
```
Comment (mashikag) [inline projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/subscribers.php]: Would it make sense to use plural here too? As in `subscriber_counts`? It is about semantics only.

**Review by mashikag — DISMISSED** [review]
The code looks fine to me.
I tested the PR and the added "include social followers" toggle seems to work as expected. 👍

mashikag jurassic tube_wp-admin_post php_post=73 action=edit calypsoify=1 (1)
mashikag jurassic tube_wp-admin_post php_post=73 action=edit calypsoify=1
Screen Shot 2022-11-22 at 11 08 13

**Review by mashikag — DISMISSED** [review]

**Review by coder-karen — COMMENTED** [review]
Testing this on a couple of test sites (Jetpack - Pressable and Jurassic Ninja) I noticed that whatever I include in the first post on the initial publish (either with or without social followers) is what shows up on any subsequent post with the subscribe block (on the front-end only - the editor shows the correct amount). I haven't worked out how to update the numbers to either show or not show social followers once that initial post has gone out. Is that expected at this stage?

**Comment by TimBroddin** [comment]
You're absolutely right. Let me look into this. 

Thanks for finding this issue @coder-karen

**Comment by TimBroddin** [comment]
There was indeed something missing. 😖 

Now tested with JT, WPCOM & Atomic:

| Jurassic Tube | WPCOM | Atomic   |
|--------------|-----------|----------|
| CleanShot 2022-11-24 at 11 48 32@2x | CleanShot 2022-11-24 at 11 48 13@2x | CleanShot 2022-11-24 at 11 54 22@2x |

**Comment by TimBroddin** [comment]
Did some refactoring after seeing that the block & shortcode widget are now [decoupled](https://github.com/Automattic/jetpack/pull/26947).

This is ready for another team review next week.

**Review by mashikag — APPROVED** [review]
I had another look and the last commit seems to be fixing the issue mentioned by @coder-karen .  All looks good to me. 👍

**Review by bindlegirl — APPROVED** [review]
LGTM

**Pull request merged**